packages/protocol/contracts/bridge/erc721/ERC721Vault.sol

```diff
@@ -0,0 +1,379 @@
+     * destination chain so the user can receive the same (bridged) tokens
+     * by invoking the message call.
+     *
+     * @param opt Option for sending the ERC721/ERC1155 token.
+     */
+    function sendToken(BridgeTransferOp calldata opt) external payable nonReentrant {
+        if (
+            opt.to == address(0) ||
+            opt.to == resolve(opt.destChainId, "erc721_vault", false)
+        ) revert NFTVAULT_INVALID_TO();
+
+        if (opt.token == address(0)) revert NFTVAULT_INVALID_TOKEN();
+
+        if (opt.amount != 1) revert NFTVAULT_INVALID_AMOUNT();
+
+        bytes memory data = _sendToken(
+            msg.sender,
+            opt.to,
+            opt.tokenId,
+            opt.token,
+            opt.baseTokenUri,
+            isBridgedToken[opt.token],
+            bridgedToCanonical[opt.token],
+            ERC721Vault.receiveToken.selector
+        );
+    
+        IBridge.Message memory message;
+        message.destChainId = opt.destChainId;
+        message.owner = msg.sender;
+        message.to = resolve(opt.destChainId, "erc721_vault", false);
+        message.data = data;
+        message.gasLimit = opt.gasLimit;
+        message.processingFee = opt.processingFee;
+        message.depositValue = 0;
+        message.refundAddress = opt.refundAddress;
+        message.memo = opt.memo;
+
+        bytes32 msgHash = IBridge(resolve("bridge", false)).sendMessage{
+            value: msg.value
+        }(message);
+
+        emit ERC721Sent({
+            msgHash: msgHash,
+            from: message.owner,
+            to: opt.to,
+            destChainId: opt.destChainId,
+            token: opt.token,
+            tokenId: opt.tokenId
+        });
+    }
+
+    /**
+     * @dev This function can only be called by the bridge contract while
+     * invoking a message call. See sendToken, which sets the data to invoke
+     * this function.
+     * @param canonicalToken The canonical ERC721 token which may or may not
+     * live on this chain. If not, a BridgedERC721 contract will be
+     * deployed.
+     * @param from The source address.
+     * @param to The destination address.
+     * @param tokenId The tokenId to be sent.
+     */
+    function receiveToken(
+        CanonicalNFT calldata canonicalToken,
+        address from,
+        address to,
+        uint256 tokenId
+    ) external nonReentrant onlyFromNamed("bridge") {
+        address bridgedAddress = canonicalToBridged[canonicalToken.srcChainId][canonicalToken.tokenAddr];
+        (
+            bool bridged,
+            address bridgedToken
+        ) = _receiveToken(
+                address(_addressManager),
+                canonicalToken,
+                from,
+                to,
+                tokenId,
+                bridgedAddress
+            );
+
+        if (bridged) {
+            setBridgedToken(bridgedToken, canonicalToken);
+        }
+    }
+
+    function releaseToken(
```

Comment:
I noticed you don't have a `messageDeposits` defined in this contract, will this releaseToken function work without it? Please read ERC20 vault.